### PR TITLE
feat!: Move arrays from prelude into new extension

### LINKED ISSUE #1497: Move Array ops to a dedicated extension
I'm not sure we've got the new array ops naming quite right. The operation names (mostly) do not mention array  This would be fine if they were in an array extension, but in prelude we now have `prelude.get`, `prelude.set` . These are very general names for ops that only work on a specific type. I suggest creating an `array` , or `prelude.array` extension.

### PR DESCRIPTION
**Pull request by aborgna-q**
Moves the prelude array to a new `collections.array` extension.
This does not yet include the array value requested in https://github.com/CQCL/hugr/issues/1497#issuecomment-2527487618, I'll make a separate PR for that.

Closes #1497

BREAKING CHANGE: Array type and operations have been moved out of `prelude` and into a new `collections.array` extension.
BREAKING CHANGE: (py) `list_type` method replaced with `List` class.
BREAKING CHANGE: Removed `Array` type variant from the serialization format.

### CODE REVIEW

**Comment by aborgna-q** [comment]
I may be going crazy, but this line in the failing `clippy` check does not exist?

File: hugr-py/src/hugr/std/collections/array.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+EXTENSION = _load_extension("collections.array")
+
+
+@dataclass(eq=False)
+class Array(tys.ExtType):
+    """Fixed `size` array of `ty` elements."""
+
+    ty: tys.Type = field(default_factory=lambda: tys.Unit)
+    size: int = field(default=0)
```
Comment (ss2165) [inline hugr-py/src/hugr/std/collections/array.py:18]: i think this stops a type arg being used here?

File: hugr-py/src/hugr/std/collections/list.py [hunk 2]
```diff
@@ -12,10 +12,22 @@
 EXTENSION = _load_extension("collections.list")
 
 
-def list_type(ty: tys.Type) -> tys.ExtType:
-    """Returns a list type with a fixed element type."""
-    arg = tys.TypeTypeArg(ty)
-    return EXTENSION.types["List"].instantiate([arg])
+@dataclass(eq=False)
+class List(tys.ExtType):
```
Comment (ss2165) [inline hugr-py/src/hugr/std/collections/list.py:16]: needs a test

File: hugr-py/src/hugr/std/collections/array.py (hunk 1 above)
Comment (aborgna-q) [reply hugr-py/src/hugr/std/collections/array.py:18]: Ah, yes. The parameter should be a `TypeArg` instead, and we cannot store a concrete type.

I'll add some tests

File: hugr-llvm/src/emit/test.rs [hunk 3]
```diff
@@ -733,7 +731,7 @@ mod test_fns {
     ) {
         exec_ctx.add_extensions(add_int_extensions);
         assert_eq!(
-            input * 1 << (iters + 1),
+            input << (iters + 1),
```
Comment (ss2165) [inline hugr-llvm/src/emit/test.rs]: these should be in main now

**Review by ss2165 — APPROVED** [review]
:shipit:

**Pull request merged**